test/e2e/suites/e2e/e2e_test.go

```diff
@@ -445,33 +452,131 @@ var _ = Describe("e2e tests [PR-Blocking]", func() {
 			Expect(err).NotTo(HaveOccurred())
 			Expect(securityGroupsList).To(HaveLen(1))
+
+			// Testing subports
+			shared.Logf("Create a new port and add it as a subport of the trunk")
+
+			providerClient, clientOpts, _, err := shared.GetTenantProviderClient(e2eCtx)
+			shared.Logf("providerClient: %+v", providerClient)
+			shared.Logf("clientOpts: %+v", clientOpts)
+			Expect(err).To(BeNil(), "Cannot create providerClient")
+
+			networkClient, err := openstack.NewNetworkV2(providerClient, gophercloud.EndpointOpts{
+				Region: clientOpts.RegionName,
+			})
+			Expect(err).To(BeNil(), "Cannot create network client")
+
+			networksList, err := shared.DumpOpenStackNetworks(e2eCtx, networks.ListOpts{})
+			Expect(err).To(BeNil(), "Cannot get network List")
+
+			createOpts := ports.CreateOpts{
+				Name:		"subPort",
+				NetworkID:      networksList[0].ID,
+				SecurityGroups: &[]string{securityGroupsList[0].ID},
+			}
+
+			subPort, err := ports.Create(ctx, networkClient, createOpts).Extract()
+			Expect(err).To(BeNil(), "Cannot create subPort")
+
+			addSubportsOpts := trunks.AddSubportsOpts{
+				Subports: []trunks.Subport{
+					{
+						SegmentationID:   1,
+						SegmentationType: "vlan",
+						PortID:           subPort.ID,
+					},
+				},
+			}
+			_, err = trunks.AddSubports(ctx, networkClient, trunk.ID, addSubportsOpts).Extract()
+			Expect(err).To(BeNil(), "Cannot add subports")
+
+			subports, err := trunks.GetSubports(ctx, networkClient, trunk.ID).Extract()
+			Expect(err).To(BeNil())
+			Expect(subports).To(HaveLen(1))
+
+			c := e2eCtx.Environment.BootstrapClusterProxy.GetClient()
+
+			machines := framework.GetMachinesByMachineDeployments(ctx, framework.GetMachinesByMachineDeploymentsInput{
+				Lister:            c,
+				ClusterName:       clusterName,
+				Namespace:         namespace.Name,
+				MachineDeployment: *machineDeployment,
+			})
+
+			Expect(machines).To(HaveLen(1))
+
+			machine := machines[0]
+
+			serverID, err := getServerIDFromMachineName(ctx, c, machine.Namespace, machine.Name)
+			Expect(err).To(BeNil())
+
+			shared.Logf("Deleting the machine deployment, which should trigger trunk deletion")
+
+			err = c.Delete(ctx, machineDeployment)
+			Expect(err).To(BeNil())
+
+			mdNs := machineDeployment.Namespace
+			mdName := machineDeployment.Name
+			Eventually(
+				func() error {
+					err := c.Get(ctx, crclient.ObjectKey{Namespace: mdNs, Name: mdName}, machineDeployment)
+					return err
+				}, e2eCtx.E2EConfig.GetIntervals(specName, "wait-machine-status")...,
+			).Should(Not(BeNil()))
+
+			shared.Logf("Waiting for the server to be cleaned")
+
+			computeClient, err := openstack.NewComputeV2(providerClient, gophercloud.EndpointOpts{
+				Region: clientOpts.RegionName,
+			})
+			Expect(err).To(BeNil(), "Cannot create compute client")
+
+			Eventually(
+				func() bool {
+					_, err := servers.Get(ctx, computeClient, serverID).Extract()
```

Comment:
Do we really need to fetch the Server again, if the Machine is gone (at line 551)?

Reply:
Thanks. Good point, I guess maybe just waiting for the Server gone would be enough.